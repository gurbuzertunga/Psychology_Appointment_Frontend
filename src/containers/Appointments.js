import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import AppointmentList from '../components/appointmentList';

const Appointments = props => {
  const { authToken } = props;
  if (!authToken) {
    return <Redirect to="/" />;
  }
  const handleClick = data => console.log(data); // eslint-disable-line
  return (
    <div className="flex flex-col items-center justify-evenly">
      <h1 className="text-white lg:text-2xl sm:text-4xl text-2xl lg:mt-4 sm:mt-24 mt-12 font-bold">Appointments you have</h1>
      <AppointmentList handleClick={handleClick} />
    </div>
  );
};

Appointments.propTypes = {
  authToken: PropTypes.string,
};
Appointments.defaultProps = {
  authToken: '',
};

const mapStateToProps = state => ({
  authToken: state.authenticationReducer.authToken,
});

export default connect(mapStateToProps)(Appointments);
