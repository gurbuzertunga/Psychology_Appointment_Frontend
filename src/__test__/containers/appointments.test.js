import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Appointments from '../../containers/Appointments';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Appointments />
  </Provider>,
);

describe('Appointments Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.appointmentList');
    expect(page).toMatchSnapshot();
  });
});
