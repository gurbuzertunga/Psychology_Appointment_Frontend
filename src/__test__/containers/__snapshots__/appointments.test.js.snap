// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
